Fixes #1753: Redux integration for Forgot password component

AakashMallik opened the pull request:
Fixes #1753 

Changes: The changes I made are:
- [x] Grouping imports in a logical fashion
- [x] Renaming variables so as to reduce the need for comments
- [x] Make functions simpler wherever possible
- [x] Put styles out of the component
- [x] Convert normal functions to arrow functions
- [x] Move propType validation inside class component.
- [x] Integrated redux for this component. 

Demo Link: https://pr-1755-fossasia-susi-web-chat.surge.sh

akshatnitd commented on src/components/Auth/ForgotPassword/ForgotPassword.react.js:
```diff
@@ -1,234 +1,132 @@
 
 class ForgotPassword extends Component {
+  static propTypes = {
+    onLoginSignUp: PropTypes.func,
```
Is this prop being used ?

akshatnitd commented on src/components/Auth/ForgotPassword/ForgotPassword.react.js:
```diff
@@ -1,234 +1,132 @@
-          if (jsonValue === 404) {
-            msg = 'Email does not exist';
+    if (email && !emailErrorMessage) {
+      this.setState({ loading: true });
```
Not required.

akshatnitd commented on src/components/Auth/ForgotPassword/ForgotPassword.react.js:
```diff
@@ -1,234 +1,132 @@
+            dialogMessage: 'Failed. Try Again',
+          });
+        });
     }
```
Make loading false here

AakashMallik replied:
I have shifted {loading: true} inside the if block. So I guess {loading: false} outside if block will not be required any more

akshatnitd commented on src/components/Auth/ForgotPassword/ForgotPassword.react.js:
```diff
@@ -1,234 +1,132 @@
-
-/* Utils*/
-import $ from 'jquery';
 import UserPreferencesStore from '../../../stores/UserPreferencesStore';
```
Is it required now?

AakashMallik replied:
User preference is being used to get themes colors in render function for now. If it had not been used, Eslint would have pointed it out.

AakashMallik commented:
@akshatnitd Sorry sir. I had my focus on integrating redux so I guess I missed those minor defects in code. I have made the changes you asked for and have proposed some counter-proposal for a few more. Let me know what you think. Thank you for your time.

AakashMallik commented on src/redux/actionTypes/index.js:
```diff
@@ -1,3 +1,7 @@
 
-export default _.mapKeys(['APP_GET_LOGIN', 'APP_GET_API_KEYS']);
+export default _.mapKeys([
+  'APP_GET_FORGOT_PASSOWORD',
```
I feel embarrassed for this typo.

AakashMallik commented:
@akshatnitd Removed the typo and unused reducer.

AakashMallik commented:
![ezgif-1-3f47fcbe70c8](https://user-images.githubusercontent.com/19551058/48969744-000d5800-f029-11e8-8755-3cb23b420e2d.gif)

@akshatnitd I have added a loader and also got rid of the pop up for error messages. I hope you like it.

AakashMallik commented:
@akshatnitd Sir, putting the loaded as a child component of material ui button did not go well so I removed it. Rest of the things have been implemented just the way you asked. And I have changed the deploy link, it was outdated.  Thanks for your time.

AakashMallik commented:
@akshatnitd Done :innocent: Changed the with of the reset button to make it look prominent. Fixed the alignment as you asked.
